RecoPixelVertexing/PixelTriplets/src/CAHitQuadrupletGenerator.cc

```diff
@@ -48,6 +48,25 @@ void CAHitQuadrupletGenerator::fillDescriptions(edm::ParameterSetDescription& de
   desc.add<bool>("useBendingCorrection", false);
   desc.add<double>("CAThetaCut", 0.00125);
   desc.add<double>("CAPhiCut", 10);
+
+  edm::ParameterSetDescription descCACut_byLayers;
+  descCACut_byLayers.add<double>("BPix1__BPix2__BPix3", -1.);
+  descCACut_byLayers.add<double>("BPix1__BPix2__FPix1_neg", -1.);
+  descCACut_byLayers.add<double>("BPix1__BPix2__FPix1_pos", -1.);
+  descCACut_byLayers.add<double>("BPix1__FPix1_neg__FPix2_neg", -1.);
+  descCACut_byLayers.add<double>("BPix1__FPix1_pos__FPix2_pos", -1.);
+  descCACut_byLayers.add<double>("BPix2__BPix3__BPix4", -1.);
+  descCACut_byLayers.add<double>("BPix2__BPix3__FPix1_neg", -1.);
+  descCACut_byLayers.add<double>("BPix2__BPix3__FPix1_pos", -1.);
+  descCACut_byLayers.add<double>("BPix2__FPix1_neg__FPix2_neg", -1.);
+  descCACut_byLayers.add<double>("BPix2__FPix1_pos__FPix2_pos", -1.);
+  descCACut_byLayers.add<double>("BPix3__FPix1_neg__FPix2_neg", -1.);
+  descCACut_byLayers.add<double>("BPix3__FPix1_pos__FPix2_pos", -1.);
+  descCACut_byLayers.add<double>("FPix1_neg__FPix2_neg__FPix3_neg", -1.);
+  descCACut_byLayers.add<double>("FPix1_pos__FPix2_pos__FPix3_pos", -1.);
```

Comment:
I would personally (even if that doesn't matter) see some value in having one syntax for layer combinations here and elsewhere.

The seeding layer configuration uses `BPix1+BPix2+BPix3`, which can't be used as parameter names (without `setattr()` etc). An alternative, although more verbose, way here would be to
```python
CAThetaCut_byLayers = cms.VPSet(
   PSet(layers = cms.string("BPix1+BPix2+BPix3"), cut = cms.double(-1.)),
   ...
)
```
The `VPSet` approach actually be extendable more easily to e.g. phase2, because "hardcoding" the possible layer combinations in `fillDescriptions()` prevents the use of any other layer combination, which feels a bit artificial restriction.